sdks/python/apache_beam/testing/util.py

```diff
@@ -108,19 +108,23 @@ def matcher(elements):
   return matcher
 
 
-# Note that equal_to always sorts the expected and actual since what we
-# compare are PCollections for which there is no guaranteed order.
-# However the sorting does not go beyond top level therefore [1,2] and [2,1]
-# are considered equal and [[1,2]] and [[2,1]] are not.
+# Note that equal_to checks if expected and actual are permutations of each
+# other. However, only permutations of the top level are checked. Therefore
+# [1,2] and [2,1] are considered equal and [[1,2]] and [[2,1]] are not.
 def equal_to(expected):
-  expected = list(expected)
 
   def _equal(actual):
-    sorted_expected = sorted(expected)
-    sorted_actual = sorted(actual)
-    if sorted_expected != sorted_actual:
+    expected_list = list(expected)
+    for element in actual:
+      try:
+        expected_list.remove(element)
+      except ValueError:
+        raise BeamAssertException(
+            'Failed assert: %r == %r' % (expected, actual))
```

Comment:
It was quite useful to have ordered elements in the message. I still think we should try to sort, and that failing fall back to a slower method.

Reply:
So try the old method first, and if it fails, falling back on the method in this PR?
That would also keep behavior on Python 2 the same.